x/proxyevm/keeper/msg_server.go

```diff
@@ -0,0 +1,154 @@
+package keeper
+
+import (
+	context "context"
+	"encoding/json"
+	"strconv"
+
+	"github.com/armon/go-metrics"
+	"github.com/cosmos/cosmos-sdk/telemetry"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	"github.com/ethereum/go-ethereum/common"
+	evmtypes "github.com/evmos/ethermint/x/evm/types"
+	tmbytes "github.com/tendermint/tendermint/libs/bytes"
+	tmtypes "github.com/tendermint/tendermint/types"
+	"github.com/xpladev/xpla/x/proxyevm/types"
+)
+
+type msgServer struct {
+	Keeper
+}
+
+// NewMsgServerImpl returns an implementation of the reward MsgServer interface
+// for the provided Keeper.
+func NewMsgServerImpl(keeper Keeper) types.MsgServer {
+	return &msgServer{Keeper: keeper}
+}
+
+var _ types.MsgServer = msgServer{}
+
+func (k msgServer) CallEVM(goCtx context.Context, msg *types.MsgCallEVM) (*evmtypes.MsgEthereumTxResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	txIndex := k.evmKeeper.GetTxIndexTransient(ctx)
+
+	var labels []metrics.Label
+	sender, err := sdk.AccAddressFromBech32(msg.Sender)
```

Comment:
think about address check position

Reply:
https://github.com/xpladev/xpla/pull/51/files#diff-eb144f638aa2d9b38e9b922e996317d447e2bd853623b03c973cce548848fbc7R37

Already checked in `ValidateBasic`. So change to must